tests/utils/refactoring_checks.py

```diff
@@ -0,0 +1,407 @@
+#!/usr/bin/env/python3
+"""This is a test script for creating a list of expected outcomes (before refactoring);
+then, manual editing might change YAMLs and/or code; another test runs to compare results
+(after refactoring to before). The target is a list of known HF repos.
+
+The goal is to identify to which extent changes break existing functionality.
+Then, larger changes to code base can be rolled out more assured.
+
+Tested with dependencies:
+pip install huggingface_hub==0.10.1 datasets==2.6.1 transformers==4.23.1
+
+dependencies as of: Oct 11, 2022; Oct 14, 2022; Oct 11, 2022.
+
+Authors
+ * Andreas Nautsch, 2022
+"""
+
+import os
+import sys
+from tqdm import tqdm
+import yaml
+import torch  # noqa
+import importlib  # noqa
+import subprocess
+import speechbrain  # noqa
+from glob import glob
+from copy import deepcopy
+from torch.utils.data import DataLoader
+from hyperpyyaml import load_hyperpyyaml
+from speechbrain.utils.distributed import run_on_main  # noqa
+from speechbrain.utils.train_logger import FileTrainLogger
+from speechbrain.pretrained.interfaces import foreign_class  # noqa
+from speechbrain.dataio.dataloader import LoopedLoader, make_dataloader
+
+
+def init(new_interfaces_git, new_interfaces_branch, new_interfaces_local_dir):
```

Comment:
Yep. For here, this is about setting up an alternative git repo within tests/tmp/... - and it might be easier if the tool handles that - can also put the lines with instruction into a readme, but I have hesitations there